### PR TITLE
mode

### PR DESCRIPTION
**Pull request by Lancelot03**

### CODE REVIEW

**Comment by Lancelot03** [comment]
#13241

**Comment by Lancelot03** [comment]
@AnnaTz  please review and suggest me the changes if any

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py [hunk 1]
```diff
@@ -797,3 +797,34 @@ def test_torch_logsumexp(
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
+        dim=axis,
+        keepdim=keepdims,
+    )
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:830]: The code seems alright, but the tests are failing so there must be some bug you'll need to solve. You can see the error in the intelligent-tests-pr/run_tests_1 CI checks. I would suggest you run the test locally to debug it.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py (hunk 1 above)
Comment (Lancelot03) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:830]: Yeah I already checked the error it was throwing pynvml warning but I already Installed it.
Screenshot 2023-04-11 000454

Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:830]: How about this 
image

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py [hunk 2]
```diff
@@ -834,3 +834,35 @@ def test_torch_unique(
         return_counts=return_counts,
         dim=axis,
     )
+    
+ 
+@handle_frontend_test(
+    fn_tree="torch.mode",
+    dtype_and_x=helpers.dtype_and_values(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:841]: I think what you should be calling here is `dtype_values_axis`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py [hunk 3]
```diff
@@ -834,3 +834,35 @@ def test_torch_unique(
...
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=-1,
+        valid_axis=True,
+        force_int_axis=True,
+    ),
+    keepdim=st.booleans(),
+)
+def test_torch_mode(
+    *,
+    dtype_input_axis,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:851]: You forgot to change this to `dtype_x_axis`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/reduction_ops.py [hunk 4]
```diff
@@ -251,3 +251,16 @@ def unique(input, sorted=True, return_inverse=False, return_counts=False, dim=No
         values.append(results.counts)
 
     return Results(*values)
+
+
+@to_ivy_arrays_and_back
+def mode(input, dim=-1, keepdim=False, *, out=None):
+    sorted_data = sorted(input.items(), key=lambda x: x[1], reverse=True)
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/reduction_ops.py:258]: There is no attribute `items`. 
You can use ivy functions in the frontends. Using `ivy.sort` directly on `input` would make things much easier.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/reduction_ops.py [hunk 5]
```diff
@@ -251,3 +251,11 @@ def unique(input, sorted=True, return_inverse=False, return_counts=False, dim=No
         values.append(results.counts)
 
     return Results(*values)
+
+
+@to_ivy_arrays_and_back
+def mode(input, dim=-1, keepdim=False, *, out=None):
+    unique_data, counts = ivy.unique(input, return_counts=True, dim=dim)
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/reduction_ops.py:258]: There is no `unique` function in our api. We've got a `unique_all` function which you can read about in our docs. This is causing an error which you should be able to see if you run the test locally.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/reduction_ops.py (hunk 5 above)
Comment (Lancelot03) [reply ivy/functional/frontends/torch/reduction_ops.py:258]: Sorry, I was taking help from the torch doc.

File: ivy/functional/frontends/torch/reduction_ops.py [hunk 6]
```diff
@@ -251,3 +251,22 @@ def unique(input, sorted=True, return_inverse=False, return_counts=False, dim=No
         values.append(results.counts)
 
     return Results(*values)
+
+
+@to_ivy_arrays_and_back
+def mode(input, dim=-1, keepdim=False, *, out=None):
+    freq = {}
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/reduction_ops.py:258]: You are not using `dim` and `keepdim` anywhere currently. I don't think the tests can be passing like this.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**